tutorials/02_build_a_scalable_question_answering_system.ipynb

```diff
@@ -0,0 +1,501 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "3. Wait 30s to ensure that the server has fully started up."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "time.sleep(30)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "4. Initialize the [`ElasticsearchDocumentStore`](https://docs.haystack.deepset.ai/reference/document-store-api#module-elasticsearch).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from haystack.document_stores import ElasticsearchDocumentStore\n",
+    "\n",
+    "# Get the host where Elasticsearch is running, default to localhost\n",
+    "host = os.environ.get(\"ELASTICSEARCH_HOST\", \"localhost\")\n",
+    "\n",
+    "document_store = ElasticsearchDocumentStore(\n",
+    "    host=host,\n",
+    "    username=\"\",\n",
+    "    password=\"\",\n",
+    "    index=\"document\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Indexing Documents with a Pipeline\n",
+    "\n",
+    "The indexing pipeline turns your files into Document objects and writes them to the DocumentStore. Our indexing pipeline will have two nodes: `TextConverter` which turns `.txt` files into Haystack `Document` objects and `PreProcessor` which cleans and splits the text within a `Document`.\n",
+    "\n",
+    "Once these nodes are combined into a pipeline, the pipeline will ingest `.txt` file paths, preprocess them, and write them into the DocumentStore.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Download 517 articles from the Game of Thrones Wikipedia. You can find them in `data/build_a_scalable_question_answering_system` as a set of `.txt` files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from haystack.utils import fetch_archive_from_http\n",
+    "\n",
+    "doc_dir = \"data/build_a_scalable_question_answering_system\"\n",
+    "\n",
+    "fetch_archive_from_http(\n",
+    "    url=\"https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/wiki_gameofthrones_txt1.zip\",\n",
+    "    output_dir=doc_dir\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. Initialize the pipeline, TextConverter, and PreProcessor."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from haystack import Pipeline\n",
+    "from haystack.nodes import TextConverter, PreProcessor\n",
+    "\n",
+    "indexing_pipeline = Pipeline()\n",
+    "text_converter = TextConverter()\n",
+    "preprocessor = PreProcessor(\n",
+    "    clean_whitespace=True,\n",
```

Comment:
Hello! Sorry if I sound a little rude. I just want to help…
As far as I understand, this tutorial should be similar to tutorial 1 and introduce the use of Elasticsearch to create a production ready system.
What is the rationale behind introducing preprocessing here? Why we don't use TextIndexingPipeline here as well?

Reply:
Hey! Am very happy to answer questions. In this case, we think its quite valuable to teach users how to build an indexing pipeline. Currently there are not so many examples of indexing pipelines, and we are trying to phase out preprocessing utility functions that we used to have.